internal/agent/device/config/controller.go

```diff
@@ -30,12 +33,31 @@ func (c *Controller) Sync(desired *v1alpha1.RenderedDeviceSpec) error {
 	c.log.Debug("Syncing device configuration")
 	defer c.log.Debug("Finished syncing device configuration")
 
-	if desired.Config == nil {
-		c.log.Debug("Device config is nil")
-		return nil
+	if desired.Config.Hooks == nil {
+		c.log.Debug("Device resources are nil")
+		// Reset all resource alerts to default
+		if err := c.hookManager.ResetDefaults(); err != nil {
+			return err
+		}
+	} else {
+		// order is important here install new hooks before applying config data
+		// so they can be consumed.
+		hooks := *desired.Config.Hooks
+		if err := c.ensureHooks(&hooks); err != nil {
```

Comment:
Why pass pointer to slice ? you can send the slice itself and then dereference will not be needed.

Reply:
yeah I am not sure why honestly I was dealing with some loops where this was required in another area. I blame old age. :) fixed.